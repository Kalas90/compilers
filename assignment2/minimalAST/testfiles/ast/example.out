Program
  List
    FunctionDec
      IdDecl
      List
      CompoundStmt
        List
