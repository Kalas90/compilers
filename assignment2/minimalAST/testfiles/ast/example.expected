Program
  List
    FunctionDec
      IdDecl
      List
        IdDecl
        IdDecl
      CompoundStmt
        List
          WhileStmt
            NEqual
              IdUse
              IdUse
            CompoundStmt
              List
                IfStmt
                  Greater
                    IdUse
                    IdUse
                  CompoundStmt
                    List
                      AssignStmt
                        IdUse
                        Sub
                          IdUse
                          IdUse
                  CompoundStmt
                    List
                      AssignStmt
                        IdUse
                        Sub
                          IdUse
                          IdUse
          ReturnStmt
            IdUse
    FunctionDec
      IdDecl
      List
        IdDecl
        IdDecl
      CompoundStmt
        List
          IfStmt
            Equal
              IdUse
              IntUse
            CompoundStmt
              List
                ReturnStmt
                  IdUse
            CompoundStmt
              List
          ReturnStmt
            FunctionFactor
              IdUse
              List
                IdUse
                Mod
                  IdUse
                  IdUse
    Comment
    FunctionDec
      IdDecl
      List
      CompoundStmt
        List
          DeclStmt
            IdDecl
            Opt
          DeclStmt
            IdDecl
            Opt
          AssignStmt
            IdUse
            FunctionFactor
              IdUse
              List
          AssignStmt
            IdUse
            FunctionFactor
              IdUse
              List
          FunctionStmt
            FunctionFactor
              IdUse
              List
                FunctionFactor
                  IdUse
                  List
                    IdUse
                    IdUse
          FunctionStmt
            FunctionFactor
              IdUse
              List
                FunctionFactor
                  IdUse
                  List
                    IdUse
                    IdUse
          DeclStmt
            IdDecl
            Opt
              Sub
                FunctionFactor
                  IdUse
                  List
                    IdUse
                    IdUse
                FunctionFactor
                  IdUse
                  List
                    IdUse
                    IdUse
          FunctionStmt
            FunctionFactor
              IdUse
              List
                IdUse
          ReturnStmt
            IntUse
